doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,431 @@
+
+
+.. _asym-verify:
+
+asymmetric_verify(*algorithm, digest_str, sig_str, pub_key_str, digest_type*)
+--------------------------------------------------------------------------------
+
+Verifies whether the signature string matches the digest string.
+
+.. rubric:: Returns
+
+A ``1`` (success) or a ``0`` (failure).
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA' or 'DSA'.
+
+* digest_str - the digest binary string and this string can be generated by invoking :ref:`create-digest<create-digest>`.
+
+* sig_str - the signature binary string. This string can be generated by invoking :ref:`asymmetric_sign<asym-sign>`.
+
+* pub_key_str - the signer's public key string. This string must correspond to the private key passed to :ref:`asymmetric_sign<asym-sign>` to generate the signature string. The string must be in the PEM format.
+
+* digest_type - the supported values are listed in the :ref:`create-digest<create-digest>` digest type table.
+
+.. _create-priv-key:
+
+create_asymmetric_priv_key(*algorithm, (key_len | dh_parameters)*)
+--------------------------------------------------------------------------------
+
+Generates a private key using the given algorithm and key length for RSA or DSA or Diffie-Hellman parameters (for DH).
+
+.. rubric:: Returns
+
+The key as a string in PEM format.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA','DSA', or 'DH'.
+
+* key_len - the supported key length values are the following:
+
+  - RSA - the minimum length is 1,024. The maximum length is 16,384.
+
+  - DSA - the minium length is 1,024. The maximum length is 9,984.
+
+    .. note:: The key length limits are defined by OpenSSL. To change the maximum key length, use either :ref:`rsa-threshold` or :ref:`dsa-threshold`.
+
+* dh_parameters - a Diffie-Hellman (DH) parameter. You can create DH parameters with the :ref:`create-dh-parameter<create-dh-param>`.
+
+.. _create-pub-key:
+
+create_asymmetric_pub_key(*algorithm, priv_key_str*)
+--------------------------------------------------------------------------
+
+Derives a public key from the given private key using the given algorithm.
+
+.. rubric:: Returns
+
+The key as a string in PEM format.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA','DSA', or 'DH'.
+
+* priv_key_str - must be a valid key string in the PEM format.
+
+.. _create-dh-param:
+
+create_dh_parameters(*key_len*)
+--------------------------------------
+
+Creates parameters for generating a Diffie-Hellman (DH) private/public key pair.
+
+.. rubric:: Returns
+
+A string in PEM format and can be passed to :ref:`create-asymmetric-private-key<create-priv-key>`.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* key_len - the range for the key length is from 1024 to 10000. The default value is 10000.
```

Comment:
Maybe use "commas" for numbering here as well for consistency. As on line no. 187,185

Reply:
Done.